### PR TITLE
Decoupling: Assemble document sidebar in `wp-story-editor`

### PR DESCRIPTION
**Pull request by mohdsayed**
## Context

https://github.com/google/web-stories-wp/issues/9070

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Partially addresses https://github.com/google/web-stories-wp/issues/9070

### CODE REVIEW

**Comment by swissspidy** [comment]
Curious, what's the reason for ff303036886e4fe889f49b59bc2731d11973722e?

**Comment by mohdsayed** [comment]
> Curious, what's the reason for [ff30303](https://github.com/google/web-stories-wp/commit/ff303036886e4fe889f49b59bc2731d11973722e)?

I tried to pass an array first so that more tabs can be sent from the configuration but I realized that there would be a couple of issues

1. The core package is not ready for this change, it would need some refactoring.
Currently, we need to provide `refs` for each tab here

https://github.com/google/web-stories-wp/blob/07d041dde2d8f26e3cb63f0822186d16d20512a8/packages/story-editor/src/components/inspector/inspectorProvider.js#L75-L81

We cannot do that dynamically, because of the limitation that a react hook cannot be used inside a loop or conditionally, and if we provide `useRef` from the `inspectorTabs` configuration, it's not making sense.

2. We cannot allow unlimited tabs because there is not enough space for them in the sidebar, and perhaps would not even be needed.

If there is a need to create one more tab later, we may provide that option by adding one spare `additional` tab like this. 

```js
<InterfaceSkeleton
  header={<Header />}
  inspectorTabs={{
    document: {
      title: __('Document', 'web-stories'),
      Pane: DocumentPane,
    },
    additional: {
      title: __('Advert', 'web-stories'),
      Pane: AdvertPane,
    }
  }}
>
```

**Comment by mohdsayed** [comment]
Circling back to the comment in https://github.com/google/web-stories-wp/pull/9569#discussion_r738479055

> Aside: #9070 will reduce this list a bit as we move more stuff out of the core app to wp-story-editor

Looks like only these two are directly being used inside the document panels, ( there are a couple more but they are reused at other places in the core package too ) which we can move.

https://github.com/google/web-stories-wp/blob/07d041dde2d8f26e3cb63f0822186d16d20512a8/packages/story-editor/src/components/panels/document/publish/publish.js#L114-L116

File: packages/wp-story-editor/src/components/documentPane/index.js [hunk 1]
```diff
@@ -13,5 +13,31 @@
...
+  StatusPanel,
+  PageAdvancementPanel,
+  BackgroundAudioPanel,
+  TaxonomiesPanel,
+} from '@web-stories-wp/story-editor';
+
+function DocumentPane() {
+  return (
+    <>
+      <StatusPanel />
```
Comment (swissspidy) [inline packages/wp-story-editor/src/components/documentPane/index.js:36]: We should move this component to `wp-story-editor`

It already contains lots of WP specific references (e.g. "Visible to site admins & editors only").
Plus, #9502 is going to update it to accommodate for WP specific quirks (e.g. WP does not supports private drafts, so private means instantly publishing the story).

File: packages/wp-story-editor/src/components/documentPane/index.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+/**
+ * Internal dependencies
+ */
+import * as publisherLogoCallbacks from '../../api/publisherLogos';
+
+function DocumentPane() {
+  return (
+    <>
+      <StatusPanel />
+      <PublishPanel apiCallbacks={publisherLogoCallbacks} />
```
Comment (swissspidy) [inline packages/wp-story-editor/src/components/documentPane/index.js]: Would it be easier if we'd move this component to `wp-story-editor` as well? Then we don't have to pass callbacks like this.

File: packages/wp-story-editor/src/components/documentPane/index.js (hunk 1 above)
Comment (mohdsayed) [reply packages/wp-story-editor/src/components/documentPane/index.js:36]: Yes, makes sense 👍

File: packages/wp-story-editor/src/components/documentPane/index.js (hunk 2 above)
Comment (mohdsayed) [reply packages/wp-story-editor/src/components/documentPane/index.js]: yeah, on it.

**Comment by mohdsayed** [comment]
Just for info, I removed the integration tests which are no longer valid in the core editor in https://github.com/google/web-stories-wp/pull/9614/commits/286fb1cad87139c21a6d17bdf0fbfe0798989cfd . Will create another PR to compensate for it in E2E tests.

**Comment by mohdsayed** [comment]
Looks like E2E won't pass here 🙁  . Will have to skip a couple of tests that are failing on `main` branch also for me locally.

Screenshot 2021-11-08 at 2 27 40 PM
Screenshot 2021-11-08 at 2 28 51 PM

**Comment by swissspidy** [comment]
I don't see them failing in this PR? Tests also seem to be passing on `main` here on GitHub.

Could you perhaps elaborate where these are failing?

**Comment by mohdsayed** [comment]
Passed after merging latest `main` today!  😺

**Review by swissspidy — APPROVED** [review]

**Pull request merged**